reports/fennec_dashboard/summarize_csv.ipynb

```diff
@@ -360,56 +361,61 @@
     "    date_range = get_last_month_range() if report_type is \"monthly\" else get_last_week_range()\n",
     "    if start_date != None and end_date != None:\n",
     "        sd = snap_to_past_sunday(start_date) if report_type is \"weekly\"\\\n",
     "                                             else snap_to_beginning_of_month(start_date)\n",
     "        date_range = (sd, end_date)\n",
-    "\n",
-    "    # Split the submission period in chunks, so we don't run out of resources while aggregating.\n",
-    "    chunk_start = date_range[0]\n",
-    "    chunk_end = None\n",
-    "    \n",
+    "        \n",
     "    delta = date_range[1] - date_range[0]\n",
     "    print \"Running summary analysis type {} over {} days for {} to {}\"\\\n",
     "            .format(report_type, delta.days, *date_range)\n",
     "\n",
+    "    # Split the submission period in chunks, so we don't run out of resources while aggregating.\n",
+    "    date_chunks = []\n",
+    "    chunk_start = date_range[0]\n",
+    "\n",
+    "    print \"Breaking date range into chunks:\"\n",
     "    while chunk_start < date_range[1]:\n",
     "        # Compute the end of this time chunk.\n",
+    "        chunk_end = None\n",
     "        if report_type == \"monthly\":\n",
     "            chunk_end = chunk_start.replace(day=calendar.monthrange(chunk_start.year, chunk_start.month)[1])\n",
     "        else:\n",
     "            chunk_end = chunk_start + dt.timedelta(days=6)\n",
+    "        \n",
+    "        print \"* {} to {}\".format(chunk_start.strftime(\"%Y%m%d\"), chunk_end.strftime(\"%Y%m%d\"))\n",
+    "        date_chunks.append((chunk_start, chunk_end))\n",
+    "        \n",
+    "        # Move on to the next chunk, just add one day to either last month or week.\n",
+    "        chunk_start = chunk_end + dt.timedelta(days=1)\n",
     "\n",
+    "    # The results will be in a dict of the form {(os, geo, channel, date): dict, ...}.\n",
+    "    results = {}\n",
+    "    \n",
+    "    for chunk_start,chunk_end in date_chunks:\n",
     "        # Fetch the pings we need.\n",
     "        submissions_range = (chunk_start.strftime(\"%Y%m%d\"), chunk_end.strftime(\"%Y%m%d\"))\n",
     "        print \"\\nFetching pings for {} to {}\".format(*submissions_range)\n",
     "        deduped = fetch_deduped_pings(submissions_range)\n",
     "        \n",
     "        # print \"Fetched {} pings\".format(deduped.count())\n",
     "\n",
-    "        chunk_results = sc.emptyRDD()\n",
-    "        \n",
+    "        chunk_results = {}\n",
     "        for query in QUERIES:\n",
     "            print \"  * Running query over dimensions: %s\" % \", \".join(query) \n",
-    "            query_result = run_query(deduped, query, report_type)\n",
+    "            query_result = dict(run_query(deduped, query, report_type).collect())\n",
+    "            #pprint(query_result)\n",
```

Comment:
Please remove commented-out code (or uncomment it). If you remove it, you can also remove the import of `pprint` above.